cluster-provision/gocli/cmd/run.go

```diff
@@ -591,6 +604,20 @@ func run(cmd *cobra.Command, args []string) (retErr error) {
 			return err
 		}
 
+		// the scripts required for running the provider exist in the node container, in order to execute them directly on the node they must be copied to the node first
+		devNull, _ := os.Open(os.DevNull)
+		success, err = docker.Exec(cli, nodeContainer(prefix, nodeName), []string{"/bin/bash", "-c", "scp -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i vagrant.key " + fmt.Sprintf("-r /scripts vagrant@192.168.66.10%d:/home/vagrant/scripts", x+1)}, devNull)
+		if err != nil {
+			return err
+		}
+
+		// move the scripts to the same location they were in in the nodecontainer to enable command abstraction
+		for _, cmd := range []string{"sudo mkdir /scripts", "sudo cp -r /home/vagrant/scripts/* /scripts"} {
```

Comment:
Why can't we do this in previous command?

Reply:
you mean `sudo mkdir /scripts && sudo cp -r /home/vagrant/scripts/* /scripts` ?

Reply:
I mean the previous command (scp) `-r /scripts vagrant@192.168.66.10%d:/home/vagrant/scripts`  moves the scripts to the VM and here we move them within the VM. Why can't we set the location right with the scp?

Reply:
Because editing the root directory requires sudo privileges and the root user does not have an ssh key so you cant scp as him